code/modules/admin/verbs/one_click_antag.dm

```diff
@@ -341,74 +345,107 @@
 		ertemplate.polldesc = prefs["polldesc"]["value"]
 		ertemplate.enforce_human = prefs["enforce_human"]["value"] == "Yes" ? TRUE : FALSE
 		ertemplate.opendoors = prefs["open_armory"]["value"] == "Yes" ? TRUE : FALSE
+		ertemplate.leader_experience = prefs["leader_experience"]["value"] == "Yes" ? TRUE : FALSE
+		ertemplate.random_names = prefs["random_names"]["value"] == "Yes" ? TRUE : FALSE
+		ertemplate.spawn_admin = prefs["spawn_admin"]["value"] == "Yes" ? TRUE : FALSE
 
-		var/list/mob/dead/observer/candidates = pollGhostCandidates("Do you wish to be considered for [ertemplate.polldesc]?", "deathsquad", null)
-		var/teamSpawned = FALSE
-
-		if(candidates.len > 0)
-			//Pick the (un)lucky players
-			var/numagents = min(ertemplate.teamsize,candidates.len)
-
-			//Create team
-			var/datum/team/ert/ert_team = new ertemplate.team
-			if(ertemplate.rename_team)
-				ert_team.name = ertemplate.rename_team
-
-			//Asign team objective
-			var/datum/objective/missionobj = new
-			missionobj.team = ert_team
-			missionobj.explanation_text = ertemplate.mission
-			missionobj.completed = TRUE
-			ert_team.objectives += missionobj
-			ert_team.mission = missionobj
-
-			var/list/spawnpoints = GLOB.emergencyresponseteamspawn
-			var/index = 0
-			while(numagents && candidates.len)
-				var/spawnloc = spawnpoints[index+1]
-				//loop through spawnpoints one at a time
-				index = (index + 1) % spawnpoints.len
-				var/mob/dead/observer/chosen_candidate = pick(candidates)
-				candidates -= chosen_candidate
-				if(!chosen_candidate.key)
-					continue
-
-				//Spawn the body
-				var/mob/living/carbon/human/ERTOperative = new ertemplate.mobtype(spawnloc)
-				chosen_candidate.client.prefs.copy_to(ERTOperative)
-				ERTOperative.key = chosen_candidate.key
-
-				if(ertemplate.enforce_human || !ERTOperative.dna.species.changesource_flags & ERT_SPAWN) // Don't want any exploding plasmemes
-					ERTOperative.set_species(/datum/species/human)
-
-				//Give antag datum
-				var/datum/antagonist/ert/ert_antag
-
-				if(numagents == 1)
-					ert_antag = new ertemplate.leader_role
-				else
-					ert_antag = ertemplate.roles[WRAP(numagents,1,length(ertemplate.roles) + 1)]
-					ert_antag = new ert_antag
+		var/list/spawnpoints = GLOB.emergencyresponseteamspawn
+		var/index = 0
 
-				ERTOperative.mind.add_antag_datum(ert_antag,ert_team)
-				ERTOperative.mind.assigned_role = ert_antag.name
+		if(ertemplate.spawn_admin)
+			if(!isobserver(usr))
+				to_chat(usr, "<span class='warning'>Could not spawn you in as briefing officer as you are not a ghost!</spawn>")
+			else
+				var/mob/living/carbon/human/admin_officer = new (spawnpoints[1])
+				var/chosen_outfit = usr.client?.prefs.brief_outfit || null
```

Comment:
```suggestion
				var/chosen_outfit = usr.client?.prefs?.brief_outfit
```

Reply:
Your code would've runtimed anyway if there was no client, you didn't do it right.